### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by phpfui**
Support for GitHub actions.

Ported to PHP 8.2.  You should check the serialization code.  I think it is correct and it passes all tests, but I have not really looked into if serialize is fully tested or works as expected.

Also specified PHP 7.1 and higher for Packagist.org

### CODE REVIEW

**Comment by phpfui** [comment]
You will have to enable github actions.  I updated the readme.md to remove travis reference, but the badge can only be tested once actions are enabled.

**Comment by DASPRiD** [comment]
I think this should also run the PHPCS rules.

File: composer.json [hunk 1]
```diff
@@ -14,6 +14,9 @@
         "enum",
         "map"
     ],
+    "require": {
+        "php": ">=7.1 <8.3"
```
Comment (DASPRiD) [inline composer.json]: Is there a reason for the upper limit limitation? That'd require us to constantly up this with every PHP release.

**Comment by phpfui** [comment]
I consider this best practice.  Neither of us know the future of PHP or how it might interact with this library. A newer version of PHP may introduce an error into a production website by a simple composer update command.  It is better to add the next version of PHP into the tests and then increment the highest version number in composer once you confirm is works in that version.

But happy to just convert this just support 7.1 and up.

Also your users will tell you if they need a newer version.  This serves as trip wire to see who is using the latest version of PHP.

**Comment by DASPRiD** [comment]
Well, we can always introduce a `<=` requirement when we see a PHP version breaking something which we cannot fix on our end :)

**Comment by phpfui** [comment]
Absolutely, but then you may have broken a production site.

This is simply a more conservative best practice approach.

Happy to change it.  Not my hill to die on, it is your package, but this is what I do for my packages.

Also happy to put in a PR when 8.3 ships.  I always try to stay on the latest.

**Comment by DASPRiD** [comment]
To be fair, within a major version, PHP should stay stable, so we could just limit it to < 9 for now.

**Comment by phpfui** [comment]
As for running the PHPCS rules, do you just want the tests to fail, or perform the suggested fix?  In my projects, I configure githooks to run the CS Fixer on pre-commit and then commit the files.  Seems to work well.

**Comment by phpfui** [comment]
OK, can add <9.0

**Comment by DASPRiD** [comment]
> As for running the PHPCS rules, do you just want the tests to fail, or perform the suggested fix? In my projects, I configure githooks to run the CS Fixer on pre-commit and then commit the files. Seems to work well.

Just let it fail.

**Comment by phpfui** [comment]
Sounds good.

**Comment by DASPRiD** [comment]
Thanks a bunch, had to apply a few minor fixes after merge, but it's all good now! :+1:

**Comment by phpfui** [comment]
Great, happy to help.

**Pull request merged**